ui/src/reusable_ui/components/dropdowns/MultiSelectDropdown.tsx

```diff
@@ -0,0 +1,252 @@
+import DropdownButton from 'src/reusable_ui/components/dropdowns/DropdownButton'
+import FancyScrollbar from 'src/shared/components/FancyScrollbar'
+
+// Types
+import {
+  DropdownMenuColors,
+  ComponentStatus,
+  ComponentColor,
+  ComponentSize,
+  IconFont,
+} from 'src/reusable_ui/types'
+
+import {ErrorHandling} from 'src/shared/decorators/errors'
+
+interface Props {
+  children: JSX.Element[]
+  onChange: (selectedIDs: string[], value: any) => void
+  onCollapse?: () => void
+  selectedIDs: string[]
+  buttonColor?: ComponentColor
+  buttonSize?: ComponentSize
+  menuColor?: DropdownMenuColors
+  status?: ComponentStatus
+  widthPixels?: number
+  icon?: IconFont
+  wrapText?: boolean
+  customClass?: string
+  maxMenuHeight?: number
+  emptyText?: string
+  separatorText?: string
+}
+
+interface State {
+  expanded: boolean
+}
+
+@ErrorHandling
+class MultiSelectDropdown extends Component<Props, State> {
+  public static defaultProps: Partial<Props> = {
+    buttonColor: ComponentColor.Default,
+    buttonSize: ComponentSize.Small,
+    status: ComponentStatus.Default,
+    widthPixels: 120,
+    wrapText: false,
+    maxMenuHeight: 250,
+    menuColor: DropdownMenuColors.Sapphire,
+    emptyText: 'Choose an item',
+    separatorText: ', ',
+  }
+
+  public static Button = DropdownButton
+  public static Item = DropdownItem
+  public static Divider = DropdownDivider
+
+  constructor(props: Props) {
+    super(props)
+
+    this.state = {
+      expanded: false,
+    }
+  }
+
+  public render() {
+    const width = `${this.props.widthPixels}px`
+
+    this.validateChildCount()
+
+    return (
+      <ClickOutside onClickOutside={this.collapseMenu}>
+        <div className={this.containerClassName} style={{width}}>
+          {this.button}
+          {this.menuItems}
+        </div>
+      </ClickOutside>
+    )
+  }
+
+  private toggleMenu = (): void => {
+    this.setState({expanded: !this.state.expanded})
+  }
+
+  private collapseMenu = (): void => {
+    const {onCollapse} = this.props
+
+    this.setState({expanded: false})
+    onCollapse()
+  }
+
+  private get containerClassName(): string {
+    const {buttonColor, buttonSize, status, wrapText, customClass} = this.props
+
+    return classnames(
+      `dropdown dropdown-${buttonSize} dropdown-${buttonColor}`,
+      {
+        disabled: status === ComponentStatus.Disabled,
+        'dropdown-wrap': wrapText,
+        [customClass]: customClass,
+      }
+    )
+  }
+
+  private get button(): JSX.Element {
+    const {
+      selectedIDs,
+      status,
+      buttonColor,
+      buttonSize,
+      icon,
+      children,
+      emptyText,
+      separatorText,
+    } = this.props
+    const {expanded} = this.state
+
+    const selectedChildren = children.filter(child =>
+      _.includes(selectedIDs, child.props.id)
+    )
+
+    let label
+
+    if (selectedChildren.length) {
+      label = selectedChildren.map((sc, i) => {
```

Comment:
what is the type of label? it seems as though there might be some type mismatch between `emptyText` and the return from mapping over selectedChildren